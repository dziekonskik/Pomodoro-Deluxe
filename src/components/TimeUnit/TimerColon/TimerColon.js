import React from 'react';
import styles from './TimerColon.module.scss';
import { Image, Container } from 'react-bootstrap';
import dot from '../../../img/svg/dot.svg';
const TimerColon = () => (
  <Container className="w-25 d-flex flex-sm-column align-items-sm-center my-auto">
    <Image className={styles.timerColonStyles} src={dot} roundedCircle />
    <Image className={styles.timerColonStyles} src={dot} roundedCircle />
  </Container>
);

export default TimerColon;
