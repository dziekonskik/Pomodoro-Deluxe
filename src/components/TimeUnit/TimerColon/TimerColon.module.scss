.timerColonStyles {
  width: 10px;
  height: 10px;
  margin: 5px;
  animation: blink 1s linear infinite;
}

@keyframes blink {
  0% {
    transform: rotate(0);
  }
  100% {
    transform: rotate(360deg);
  }
}
