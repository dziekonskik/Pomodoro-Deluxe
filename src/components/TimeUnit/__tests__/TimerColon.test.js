import React from 'react';
import TimerColon from '../TimerColon/TimerColon';
import renderer from 'react-test-renderer';

let TimerColonRenderer;
beforeEach(() => (TimerColonRenderer = renderer.create(<TimerColon />)));

it('renders properly', () => {
  expect(TimerColonRenderer.toJSON()).toMatchSnapshot();
});

it('have exactly 2 children', () => {
  expect(TimerColonRenderer.toJSON().children.length).toBe(2);
});

it('has images as children', () => {
  expect(TimerColonRenderer.toJSON().children[0].type).toMatch('img');
  expect(TimerColonRenderer.toJSON().children[1].type).toMatch('img');
});
