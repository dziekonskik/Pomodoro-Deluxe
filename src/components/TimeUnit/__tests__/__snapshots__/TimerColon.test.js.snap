// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="d-flex flex-column align-items-center container"
>
  <img
    className="timerColonStyles rounded-circle"
    src="dot.svg"
  />
  <img
    className="timerColonStyles rounded-circle"
    src="dot.svg"
  />
</div>
`;
